src/BoundaryConditions/apply_value_gradient_bcs.jl

```diff
@@ -36,8 +36,8 @@ end
 @inline bottom_gradient(bc::VBC, c¹, Δ, i, j, args...) = ( c¹ - getbc(bc, i, j, args...) ) / (Δ/2)
 @inline    top_gradient(bc::VBC, cᴺ, Δ, i, j, args...) = ( getbc(bc, i, j, args...) - cᴺ ) / (Δ/2)
 
-@inline  left_gradient(bc::VBC, c¹, Δ, i, k, args...) = ( c¹ - getbc(bc, i, k, args...) ) / (Δ/2)
-@inline right_gradient(bc::VBC, cᴺ, Δ, i, k, args...) = ( getbc(bc, i, k, args...) - cᴺ ) / (Δ/2)
+@inline south_gradient(bc::VBC, c¹, Δ, i, k, args...) = ( c¹ - getbc(bc, i, k, args...) ) / (Δ/2)
```

Comment:
This function's definition is identical to `bottom_gradient`.

I think we should have two functions, `left_gradient` and `right_gradient`. `left_gradient` should be called in `_fill_bottom_halo!(c, bc::Union{VBC, GBC}, grid, args...)`, `_fill_south_halo!(c, bc::Union{VBC, GBC}, grid, args...)`, and `_fill_west_halo!(c, bc::Union{VBC, GBC}, grid, args...)`, while `right_gradient` should be called in the same functions for `top`, `north`, and `east`.

and 

```